tianshou/policy/imitation/discrete_crr.py

```diff
@@ -1,34 +1,36 @@
 from copy import deepcopy
-from typing import Any
+from typing import Any, Literal
 
+import gymnasium as gym
 import torch
 import torch.nn.functional as F
 from torch.distributions import Categorical
 
 from tianshou.data import to_torch, to_torch_as
 from tianshou.data.types import RolloutBatchProtocol
+from tianshou.policy.base import TLearningRateScheduler
 from tianshou.policy.modelfree.pg import PGPolicy
 
 
 class DiscreteCRRPolicy(PGPolicy):
     r"""Implementation of discrete Critic Regularized Regression. arXiv:2006.15134.
 
-    :param torch.nn.Module actor: the actor network following the rules in
+    :param actor: the actor network following the rules in
         :class:`~tianshou.policy.BasePolicy`. (s -> logits)
-    :param torch.nn.Module critic: the action-value critic (i.e., Q function)
+    :param critic: the action-value critic (i.e., Q function)
         network. (s -> Q(s, \*))
-    :param torch.optim.Optimizer optim: a torch.optim for optimizing the model.
-    :param float discount_factor: in [0, 1]. Default to 0.99.
+    :param optim: a torch.optim for optimizing the model.
+    :param discount_factor: in [0, 1]. Default to 0.99.
     :param str policy_improvement_mode: type of the weight function f. Possible
         values: "binary"/"exp"/"all". Default to "exp".
-    :param float ratio_upper_bound: when policy_improvement_mode is "exp", the value
+    :param ratio_upper_bound: when policy_improvement_mode is "exp", the value
         of the exp function is upper-bounded by this parameter. Default to 20.
-    :param float beta: when policy_improvement_mode is "exp", this is the denominator
+    :param beta: when policy_improvement_mode is "exp", this is the denominator
         of the exp function. Default to 1.
-    :param float min_q_weight: weight for CQL loss/regularizer. Default to 10.
-    :param int target_update_freq: the target network update frequency (0 if
+    :param min_q_weight: weight for CQL loss/regularizer. Default to 10.
+    :param target_update_freq: the target network update frequency (0 if
         you do not use the target network). Default to 0.
-    :param bool reward_normalization: normalize the reward to Normal(0, 1).
+    :param reward_normalization: normalize the reward to Normal(0, 1).
         Default to False.
```

Comment:
missing docstring? action_bound_method for example

Reply:
I tried my best to keep the docstrings correct, but there are probably still several incomplete ones... Gonna address the ones you found

Reply:
Done